starknet_py/net/client_models.py

```diff
@@ -226,7 +212,26 @@ class TransactionFinalityStatus(Enum):
     ACCEPTED_ON_L1 = "ACCEPTED_ON_L1"
 
 
-# TODO (#1047): split into PendingTransactionReceipt and TransactionReceipt?
+@dataclass
+class ExecutionResources:
+    """
+    Dataclass representing the resources consumed by the transaction.
+    """
+
+    # pylint: disable=too-many-instance-attributes
+
+    steps: int
+    range_check_builtin_applications: int
+    pedersen_builtin_applications: int
+    poseidon_builtin_applications: int
+    ec_op_builtin_applications: int
+    ecdsa_builtin_applications: int
+    bitwise_builtin_applications: int
+    keccak_builtin_applications: int
+    memory_holes: Optional[int] = None
+
+
+# TODO (#1179): split into PendingTransactionReceipt and TransactionReceipt
```

Comment:
Are you sure this is correct issue no? 👀

Reply:
I decided to put this issue here as these changes will be easier to do after updating devnet. Maybe we should consider creating a separate issue though.